### ISSUE TITLE
--first doesn't match from very beginning

### ISSUE THREAD

**Issue by fullmetalScience** [hinto-janai/monero-vanity#4]
`--first --pattern 44xmr` also produces other addresses like ones starting with `444xmr..`

To reproduce: `while ! monero-vanity --first --pattern 44x | grep -A99 -B99 ' 444x' ; do clear; done`

This is a very minor issue that's easily worked around by adding `^` to the pattern, but may be worth fixing at least in the docs.

**Comment by hinto-janai** [comment]
Right, what `--first` actually does is give you full control over the regex, it should be more  clear.

PR to fix is welcome.
